### ISSUE TITLE
Bug | CloudWatch logs flushes only once per app session

### ISSUE THREAD

**Issue by KoprivaMartin** [aws-amplify/amplify-swift#3199]
### Describe the bug

Logs file gets delete on flush and new file is not created so new logs are not stored and flushed.

`RotatingLogBatch.complete()` removes the file. It's called from `CloudWatchLoggingConsumer.consume()`.
I guess that there is missing logic for creating new rotating log file.

### Steps To Reproduce

```swift
Follow documentation to setup Amplify Logging, wait until the app flushes first logs batch.
Observe no other logs are created/flushed after first batch.
```

### Expected behavior

Create new logs file after batch is sent.
I assume `LogRotation.rotate()` should be called after previous file is deleted.

### Amplify Framework Version

2.16.0

### Amplify Categories

Analytics

### Dependency manager

Swift PM

### Swift version

5.8.1

### CLI version

12.3.0

### Xcode version

14.3.1 (14E300c)

### Is this a regression?

Yes

### Platforms

iOS

### OS Version

iOS 16.4

### Device

iPhone 13 Pro

**Comment by phantumcode** [comment]
@KoprivaMartin Thanks for opening the issue, we'll investigate and try to reproduce the issue.

**Comment by phantumcode** [comment]
Fix is released in [version 2.17.1](https://github.com/aws-amplify/amplify-swift/releases/tag/2.17.1)
